File: python/78 BackAccount.py
class BankAccount:
    def __init__(self,accountNumber,accountHolderName,balance):
        self.__accountNumber=accountNumber
        self.__accountHolderName=accountHolderName
        self.__balance=balance
    def getCurrentBalance(self):
        return self.__balance
    def deposit(self,amount):
        if amount>0:
            self.__balance+=amount
            print("Deposit successful")
        else:
            print("Nice try")
    def withdraw(self,amount):
        if amount>self.__balance:
            print("Low balance")
        else:
            self.__balance-=amount
            print(f"Withdrawal of ${amount:.2f} successful")

acc = BankAccount("12345", "Alice Smith", 1000.00)

print(f"\nInitial balance: ${acc.getCurrentBalance():.2f}")

acc.deposit(200.50)
acc.deposit(-50)

acc.withdraw(150.75)
acc.withdraw(1500.00)
acc.withdraw(-100)

print(f"\nFinal balance: ${acc.getCurrentBalance():.2f}")

File: java/78 BackAccount.java
package code;
class BankAccount{
	private double balance;
	private String accNum,accHolderName;
	public BankAccount(double balance, String accNum, String accHolderName) {
		this.balance = balance;
		this.accNum = accNum;
		this.accHolderName = accHolderName;
	}
	public double getBalance() {
		return balance;
	}
	public void deposit(double amount) {
		if(amount<=0) {
			System.out.println("Invalid amount");
		}
		else {
			this.balance+=amount;
			System.out.println("Balance updated");
		}
	}
	public void withdraw(double amount) {
		if(amount>this.balance || amount<=0) {
			System.out.println("Invalid amount");
		}
		else {
			this.balance-=amount;
			System.out.println("Withdrawal successful");
		}
	}
}
	
public class Test {
	public static void main(String[] args) {
		BankAccount myAccount = new BankAccount(1000.0, "SB-123456", "Alice Smith");

        myAccount.deposit(500.0);

        myAccount.deposit(-100.0);

        myAccount.withdraw(200.0);

        myAccount.withdraw(1500.0);

        myAccount.withdraw(0);
	}
}